bundle generation TEMP_FILE_PATH sometimes breaks parallel react-native builds

### Description


During bundle creation a temp file is used in _TMPDIR_/CodePushResourcesMap.json 

For concurrent executions (for us this is the case on our continuous integration server) this allows race conditions resulting in unexpected build failures and _maybe_ to other unexpected results. 

Sequence: 

First in scripts/recordFilesBeforeBundleCommand.js the file is created and written: 
```
var TEMP_FILE_PATH = path.join(require("os").tmpdir(), "CodePushResourcesMap.json");
// [ ... ]
fs.writeFile(TEMP_FILE_PATH, JSON .... 
```

later in scripts/generateBundledResourcesHash.js the file is read and removed: 
```
var TEMP_FILE_PATH = path.join(require("os").tmpdir(), "CodePushResourcesMap.json");
// [ ... ]
var oldFileToModifiedTimeMap = require(TEMP_FILE_PATH);
// [ ... ]
fs.unlinkSync(TEMP_FILE_PATH);
```

For us it happens quite often that multiple jobs are running in parallel, breaking each other. 
This is the sample output from the affected jenkins job:

```
[ ... ]
bundle: Done copying assets
:app:generateBundledResourcesHashLocalReleasemodule.js:471
    throw err;
    ^

Error: Cannot find module '/tmp/CodePushResourcesMap.json'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/srv/jenkins_home/workspace/my-feature-branch-S7T3KVECBPSJLFN6HQNB4IPKGNWPRYTU3HWBIQIZ6YVEAAKS4E6Q/node_modules/react-native-code-push/scripts/generateBundledResourcesHash.js:32:32)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
 FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:generateBundledResourcesHashLocalRelease'.
> Process 'command 'node'' finished with non-zero exit value 1
```
### Reproduction

Use gradle to build two codepush enabled react-native projects at the same time (e.g. different feature branches)

### Additional Information

* react-native-code-push version: ^1.14.6-beta (but obviously affects versions up to latest)
* react-native version: ^0.36.1
* iOS/Android/Windows version: none as this is a compile time issue, but we used android build
* Does this reproduce on a debug build or release build? Release build
* this at least happens during android gradle builds

--- a/android/codepush.gradle
+++ b/android/codepush.gradle
@@ -52,11 +52,13 @@ gradle.projectsEvaluated {
             
             def jsBundleFile = file("$jsBundleDir/$bundleAssetName")
 
+            def resourcesMapTempFileName = "CodePushResourcesMap-" + java.util.UUID.randomUUID().toString().substring(0,8) + ".json"
+
             // Make this task run right before the bundle task
             def recordFilesBeforeBundleCommand = tasks.create(
                     name: "recordFilesBeforeBundleCommand${targetName}",
                     type: Exec) {
-                commandLine "node", "${nodeModulesPath}/react-native-code-push/scripts/recordFilesBeforeBundleCommand.js", resourcesDir
+                commandLine "node", "${nodeModulesPath}/react-native-code-push/scripts/recordFilesBeforeBundleCommand.js", resourcesDir, resourcesMapTempFileName
             }
 
             recordFilesBeforeBundleCommand.dependsOn("merge${targetName}Resources")
@@ -67,7 +69,7 @@ gradle.projectsEvaluated {
             def generateBundledResourcesHash = tasks.create(
                     name: "generateBundledResourcesHash${targetName}",
                     type: Exec) {
-                commandLine "node", "${nodeModulesPath}/react-native-code-push/scripts/generateBundledResourcesHash.js", resourcesDir, "$jsBundleDir/$bundleAssetName", jsBundleDir
+                commandLine "node", "${nodeModulesPath}/react-native-code-push/scripts/generateBundledResourcesHash.js", resourcesDir, "$jsBundleDir/$bundleAssetName", jsBundleDir, resourcesMapTempFileName
             }
 
             generateBundledResourcesHash.dependsOn("bundle${targetName}JsAndAssets")

--- a/scripts/generateBundledResourcesHash.js
+++ b/scripts/generateBundledResourcesHash.js
@@ -21,16 +21,18 @@ var CODE_PUSH_FOLDER_PREFIX = "CodePush";
 var CODE_PUSH_HASH_FILE_NAME = "CodePushHash";
 var CODE_PUSH_HASH_OLD_FILE_NAME = "CodePushHash.json";
 var HASH_ALGORITHM = "sha256";
-var TEMP_FILE_PATH = path.join(require("os").tmpdir(), "CodePushResourcesMap.json");
 
 var resourcesDir = process.argv[2];
 var jsBundleFilePath = process.argv[3];
 var assetsDir = process.argv[4];
+var tempFileName = process.argv[5];
+
+var tempFileLocalPath = path.join(require("os").tmpdir(), tempFileName);
 var resourceFiles = [];
 
 getFilesInFolder(resourcesDir, resourceFiles);
 
-var oldFileToModifiedTimeMap = require(TEMP_FILE_PATH);
+var oldFileToModifiedTimeMap = require(tempFileLocalPath);
 var newFileToModifiedTimeMap = {};
 
 resourceFiles.forEach(function(resourceFile) {
@@ -114,4 +116,4 @@ function fileExists(file) {
     catch (e) { return false; }
 }
 
-fs.unlinkSync(TEMP_FILE_PATH);
+fs.unlinkSync(tempFileLocalPath);

--- a/scripts/recordFilesBeforeBundleCommand.js
+++ b/scripts/recordFilesBeforeBundleCommand.js
@@ -11,9 +11,11 @@ var path = require("path");
 
 var getFilesInFolder = require("./getFilesInFolder");
 
-var TEMP_FILE_PATH = path.join(require("os").tmpdir(), "CodePushResourcesMap.json");
 
 var resourcesDir = process.argv[2];
+var tempFileName = process.argv[3];
+
+var tempFileLocalPath = path.join(require("os").tmpdir(), tempFileName);
 var resourceFiles = [];
 
 try {
@@ -32,7 +34,7 @@ resourceFiles.forEach(function(resourceFile) {
     fileToModifiedTimeMap[resourceFile.path.substring(resourcesDir.length)] = resourceFile.mtime.getTime();
 });
 
-fs.writeFile(TEMP_FILE_PATH, JSON.stringify(fileToModifiedTimeMap), function(err) {
+fs.writeFile(tempFileLocalPath, JSON.stringify(fileToModifiedTimeMap), function(err) {
     if (err) {
         throw err;
     }

